src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: 'triangle' = 'triangle';
 
-}
+  constructor(
+    public color: 'red' | 'green' | 'blue',
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('check the length of the sides');
+    }
+
+    if (a >= (b + c) || b >= (a + c) || c >= (a + b)) {
+      throw new Error('check the length of the sides');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const perimeter: number = (this.a + this.b + this.c) / 2;
+    const area: number
+     = ((perimeter * (perimeter - this.a)
+     * (perimeter - this.b) * (perimeter - this.c))
+     ** 0.5);
```

Comment:
```suggestion
    const area: number = ((perimeter 
       * (perimeter - this.a) * (perimeter - this.b) 
       * (perimeter - this.c)) ** 0.5);
```
Looks better to me, but not a mistake